src/utils/profile.ts

```diff
@@ -0,0 +1,72 @@
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import {
+  IResponseSuccessJson,
+  ResponseErrorInternal
+} from "@pagopa/ts-commons/lib/responses";
+import { identity, pipe } from "fp-ts/lib/function";
+import { User } from "../../src/types/user";
+import ProfileService from "../../src/services/profileService";
+import { InitializedProfile } from "../../generated/backend/InitializedProfile";
+import { EmailAddress } from "../../generated/io-api/EmailAddress";
+
+// define a type that represents a Profile with a non optional email address
+const ProfileWithEmail = t.intersection([
+  t.interface({
+    email: EmailAddress
+  }),
+  InitializedProfile
+]);
+
+type ProfileWithEmail = t.TypeOf<typeof ProfileWithEmail>;
+
+// define a branded type that ensure we have the email in the profile and
+// that it is not undefined
+interface IProfileWithEmailValidatedTag {
+  readonly HasValidEmailAddress: unique symbol;
+}
+
+const ProfileWithEmailValidated = t.brand(
+  ProfileWithEmail,
+  (p): p is t.Branded<ProfileWithEmail, IProfileWithEmailValidatedTag> =>
+    !!(p.email && p.is_email_validated),
+  "HasValidEmailAddress"
+);
+
+type ProfileWithEmailValidated = t.TypeOf<typeof ProfileWithEmailValidated>;
+
+/**
+ * Gets a profile with a valid email
+ *
+ * @param profileService
+ * @param user
+ * @returns
+ */
+export const profileWithEmailValidatedOrError = (
+  profileService: ProfileService,
+  user: User
+) =>
```

Comment:
We should remove the `Response` context from this method, maybe an explicit return type declaration make it more clear.
Now the signature of this function is `TE.TaskEither<IResponseErrorInternal, ProfileWithEmailValidated>`. Maybe is better have an
```suggestion
export const profileWithEmailValidatedOrError = (
  profileService: ProfileService,
  user: User
): TE.TaskEither<Error, ProfileWithEmailValidated> =>
```

Reply:
Isn't this going to masquerade the response status code for the left side? I mean, if GetProfile returns a IResponseErrorNotFound this will be returned as a left side. Removing the response type e returning an Error, would mean "rewrapping" the left side Error into a response before the controller returns. This would cause returning very probably an IResponseErrorInternal with a message such as "User not found". Wouldn't this be semantically uncorrect?

Reply:
You are right, but just semantically speaking, is legit have a `404 Not Found` when Zendesk or any other service use a valid session token (user session is alive) with a missing user profile? If it isn't a possible scenario, when it happens the right response should be a `500 Error Internal` with a message `User not found` and it will became an issue that needs investigation.

About this particular implementation I don't want change the API interface, but the signature should became as follow to handle the not found scenario properly.
```suggestion
export const NotFoundError = new Error("User profile not found");
export const profileWithEmailValidatedOrError = (
  profileService: ProfileService,
  user: User
): TE.TaskEither<Error | NotFoundError, ProfileWithEmailValidated> =>
```
If you see more impediments we can just rename the method as `profileWithEmailValidatedOrErrorResponse` 😄

Reply:
You know? Looking at it from your point of view makes me think about it the same way. Maybe you're right and we should return 500 because it should not happen.

Reply:
Dit it in https://github.com/pagopa/io-backend/pull/926/commits/eb3ddda9555894432b373fded78c82456202b3d4

Reply:
Take care that we are changing now the API specification removing `IResponseErrorNotFound`
and `IResponseErrorTooManyRequests`. This change is safe? If is safe we need to remove the additional responses from `getZendeskSupportToken` in `zendeskController.ts` ([ref](https://github.com/pagopa/io-backend/blob/iocit-90/src/controllers/zendeskController.ts#L55)) and from `api_zendesk.yaml` (that are already missing 😄, so the implementation was inconsistent with the OpenApi specifications)

Reply:
It's safe and you're right. Done in https://github.com/pagopa/io-backend/pull/926/commits/c1697c18c8ce5f373864d6650f3f5bdee8d74ddb.